src/libraries/Microsoft.PowerFx.Connectors/OpenApiExtensions.cs

```diff
@@ -623,20 +626,45 @@ internal static ConnectorType GetConnectorType(this OpenApiParameter openApiPara
             }
         }
 
-        internal static RecordType SafeAdd(this RecordType recordType, string logicalName, FormulaType formulaType, string displayName)
+        internal static RecordType ToRecordType(this List<(string logicalName, string displayName, FormulaType type)> fields)
         {
-            int i = 0;
-            string displayName2 = displayName;
+            if (fields == null)
+            {
+                return null;
+            }
+            
+            HashSet<string> names = new HashSet<string>(fields.Select(f => f.logicalName), StringComparer.InvariantCulture);

```

Comment:
This looks like very similar logic to DisplayNameUtility.MakeUnique?

Reply:
Yes
The names that we generate are different though